### PR TITLE
MULE-15908: Persistent ObjectStore doesn't regenerate the partition d…

### PR DESCRIPTION
**Pull request by estebanwasinger**
…escriptor file after a clear

### CODE REVIEW

File: core/src/test/java/org/mule/util/store/PersistentObjectStorePartitionTestCase.java [hunk 1]
```diff
@@ -103,18 +105,18 @@ public void skipAndMoveCorruptedOrUnreadableFiles()
...
+    public void clear() throws Exception {
         final String KEY = "key";
         final String VALUE = "value";
 
 
         partition.store(KEY, VALUE);
-        assertTrue(partition.contains(KEY));
-        assertEquals(VALUE, partition.retrieve(KEY));
+        assertThat(partition.contains(KEY), is(true));
+        assertThat(VALUE, is(partition.retrieve(KEY)));
```
Comment (fsgonz) [inline core/src/test/java/org/mule/util/store/PersistentObjectStorePartitionTestCase.java]: change the order of the assertThat parameters.

File: core/src/test/java/org/mule/util/store/PersistentObjectStorePartitionTestCase.java [hunk 2]
```diff
@@ -103,18 +105,18 @@ public void skipAndMoveCorruptedOrUnreadableFiles()
...
 
         partition.clear();
-        assertFalse(partition.contains(KEY));
+        assertThat(partition.contains(KEY), is(false));
+        assertThat("Partition descriptor doesn't exists", new File(objectStoreFolder.getRoot(), "partition-descriptor").exists(), is(true));
```
Comment (fsgonz) [inline core/src/test/java/org/mule/util/store/PersistentObjectStorePartitionTestCase.java]: doesn't exist

File: core/src/test/java/org/mule/util/store/PartitionedPersistentObjectStoreTestCase.java [hunk 3]
```diff
@@ -106,6 +106,20 @@ public void clear() throws ObjectStoreException {
         this.assertAllValuesExistsInPartitionAreUnique(OBJECT_KEY, OBJECT_BASE_VALUE);
     }
 
+    @Test
+    public void clearDoesntBreakPartition() throws ObjectStoreException {
+        String partitionName = "custom";
```
Comment (fsgonz) [inline core/src/test/java/org/mule/util/store/PartitionedPersistentObjectStoreTestCase.java]: Maybe add the literals as constants.

Comment (estebanwasinger) [reply core/src/test/java/org/mule/util/store/PartitionedPersistentObjectStoreTestCase.java]: they are just used in the test

**Pull request merged**